### ISSUE TITLE
Update PlatformIO package

### ISSUE THREAD

**Issue by MattPellegrini** [esp32m/logging#15]
Searching for this library in platformio, or adding `esp32m/logging` directly to the `platformio.ini` gets quite a very old version of this library.

The fix for this is to make a commit bumping the version in `library.json` and running `pio pkg update` as per https://docs.platformio.org/en/latest/librarymanager/creating.html#publishing

For now I'm using the github link as a dependency in `platformio.ini` as a workaround to get the latest code.

Any chance of publishing the latest to platformio library manager?

**Comment by dyarkovoy** [comment]
You're right, I haven't been publishing this library to PIO for quite a while. I remember looking into it a few years back, and I found it challenging to maintain proper sync between PIO repository and github. So I decided direct github link would be so much easier. I think I tried to remove it from PIO back then to encourage using the github link, but it didn't seem possible then, so I gave up on it completely. I don't use PIO nor this library any longer and if you or anyone else would like to update PIO repository - I don't mind, let me know what's needed on my part so you can do it.

**Comment by MattPellegrini** [comment]
I gave it a try, here's what I've learned:

The process to update the package is a simple as:
1. Clone the repo to local machine
2. Update `library.json` with a version bump - I did this only locally to `1.0.4`, but it's possible this needs to be committed and pushed.
3. Open the cloned repo in VSCode with platformio extention
4. Open pio cli and `run pio pkg publish --owner esp32m` (you won't need this --owner flag).

That should publish the latest code such that people can import it only by adding `esp32m/logging @ 1.0.4` to their `platformio.ini`.

However step4 failed for me (as expected) since I don't have publishing permissions to esp32m. I can publish to `mattpellegrini/logging` of course, but that will just add confusion when searching for this library since users will see multiple options.

So we land with 3 options:
`A`. You can follow steps 1-4 above to publish a new version.
`B`. You can add my account permissions to publish new versions
`C`. I use the github URL in my platformio and the stale version remains in the package manager for unaware users (i.e. do nothing).

`B` still unfortunately requires using the pio cli, but instead of publishing a new version you should be able to run:
* `pio access list` which will should get you the URN of the package, something like this:
```
ESP32 logger
------------
URN:              prn:reg:pkg:XXXXX:library
Owner:            esp32m
Access:           Public
Access level(s):  Admin
```
* `pio access grant maintainer mattpellegrini URN` where URN is obtained above. ([docs](https://docs.platformio.org/en/latest/core/userguide/access/cmd_grant.html))

then I should be able to push to esp32m/logging.

So `B` feels like as much work as `A`, but means I can do pushes in future without bothering you. (You could also grant admin role, which then presumably lets me pass on this for others in future too).

I'm happy with `A`, `B` or `C`, it's just a matter of how up to date you want the platformio pkg manager version to be.
Let me know how I can help if you want to do `B`.

**Comment by dyarkovoy** [comment]
Thanks for the step-by-step, I went with option B, let me know if that worked:
![image](https://github.com/user-attachments/assets/87545bb0-8f26-40e8-8c8c-5f4f3e7d25a6)

**Comment by MattPellegrini** [comment]
Thanks, the version available in platformio is now 1.0.5 matching the latest [commit](https://github.com/esp32m/logging/pull/17#event-15496372655) 

Adding `esp32m/ESP32 logger@^1.0.5` works as expected and I've verified that the source code pulled down for the library is that matching 1.0.5 (library.json matches).

For posterity, in case anyone finds this thread, after updating `platformio.ini` you may need to run `pio pkg update` to refresh the dependencies and pull the latest code.
